### DISCUSSION TITLE
Travis CI and Heroku OAuth security breach

### DISCUSSION THREAD

**Discussion by wyphan** [community/community#21966] [New to GitHub]
<p>I received an email that I’m affected by a security breach published in GitHub blog:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/50928756/181067922-abae497e-ded0-4cb4-80c3-ec162a8ada8a.png" class="site-icon" width="32" height="32">

      <a href="https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/" target="_blank" rel="noopener nofollow ugc" title="10:35PM - 15 April 2022">The GitHub Blog – 15 Apr 22</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/50928756/181067923-817b7c12-d0f7-4130-99d1-3d43c87f8c6c.png" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/2/a2cee7d85cf2fc8378c296e0b5c7ff1874954bc8_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/2/a2cee7d85cf2fc8378c296e0b5c7ff1874954bc8_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/2/a2cee7d85cf2fc8378c296e0b5c7ff1874954bc8_2_1380x724.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/2/a2cee7d85cf2fc8378c296e0b5c7ff1874954bc8_2_10x10.png"></div>

<h3><a href="https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/" target="_blank" rel="noopener nofollow ugc">Security alert: Attack campaign involving stolen OAuth user tokens issued to...</a></h3>

  <p>On April 12, GitHub Security began an investigation that uncovered evidence that an attacker abused stolen OAuth user tokens issued to two third-party OAuth integrators, Heroku and Travis-CI, to download data from dozens of organizations, including...</p>

  <p>
    <span class="label1">Est. reading time: 5 minutes</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>What should I do to mitigate this?</p>
<p>I removed Travis CI from the list of installed apps at <a href="https://github.com/settings/installations" rel="noopener nofollow ugc">this page</a> since I haven’t used it for a while, is that enough?</p>

**Comment by queenofcorgis** [comment, marked as answer]
<p>Hi <a class="mention" href="/wyphan">@wyphan</a>! I would follow the blog for the latest source of truth as well as continue to monitor Heroku and Travis CI for updates on their own investigations into the affected OAuth applications at:</p>
<ul>
<li><a href="https://status.heroku.com/incidents/2413">Heroku</a></li>
<li><a href="https://blog.travis-ci.com/2022-04-17-securitybulletin">Travis CI</a></li>
</ul>
